lib/yarp/parser_gem_compat.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'ast'
+
+module YARP
+  module ParserGem
+    extend AST::Sexp
+
+    def self.parse(code, filepath = nil)
+      parse_result = YARP.parse(code, filepath)
+      if ENV["DEBUG"]
+        pp parse_result.value
+        puts
+      end
+      convert parse_result.value
+    end
+
+    def self.convert(node)
+      case node
+      when ProgramNode
+        convert node.statements
+      when StatementsNode
+        statements = node.body
+        if statements.size == 1
+          convert statements[0]
+        else
+          s(:begin, *statements.map { convert _1 })
+        end
+      when ArrayNode
+        s(:array, *node.elements.map { convert _1 })
+      when HashNode
+        s(:hash, *node.elements.map { convert _1 })
+      when AssocNode
+        s(:pair, convert(node.key), convert(node.value))
+      when AssocSplatNode
+        s(:kwsplat, convert(node.value))
+      when CallNode
+        arguments = node.arguments ? node.arguments.arguments.map { convert _1 } : []
+        receiver = node.receiver ? convert(node.receiver) : nil
+        s(:send, receiver, node.name.to_sym, *arguments)
+      when LocalVariableReadNode
+        s(:lvar, node.constant_id)
+      when LocalVariableWriteNode
+        if node.value
+          s(:lvasgn, node.constant_id, convert(node.value))
+        else
+          s(:lvasgn, node.constant_id)
+        end
+      when InstanceVariableReadNode
+        s(:ivar, node.slice.to_sym)
+      when InstanceVariableWriteNode
+        if node.value
+          s(:ivasgn, node.name.to_sym, convert(node.value))
+        else
+          s(:ivasgn, node.name.to_sym)
+        end
+      when OperatorWriteNode
+        s(:op_asgn, convert(node.target), node.operator, convert(node.value))
+      when SymbolNode
+        s(:sym, node.value.to_sym)
+      when RationalNode
+        s(:rational, node.value)
+      when ImaginaryNode
+        s(:complex, node.value)
+      when FloatNode
+        s(:float, node.value)
+      when IntegerNode
+        s(:int, node.value)
+      when TrueNode
+        s(:true)
+      when FalseNode
+        s(:false)
+      when NilNode
+        s(:nil)
```

Comment:
It would be nice to have a `LiteralNode` superclass or included module in those, and that would define a `value` method which returns the literal value.